### PR TITLE
Add support for scheduler groups and trap group workaround

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: syncd/syncd_hard_reinit.cpp [hunk 1]
```diff
@@ -717,15 +817,25 @@ sai_unicast_route_entry_t getRouteEntryFromString(const std::string &strRouteEnt
...
     SWSS_LOG_ENTER();
 
+    SWSS_LOG_TIMER("apply routes");
+
     for (auto &kv: g_routes)
     {
         const std::string &strRouteEntry = kv.first;
         const std::string &asicKey = kv.second;
 
+        bool isDefault = strRouteEntry.find("\"0.0.0.0/0\"") != std::string::npos ||
```
Comment (qiluo-msft) [inline syncd/syncd_hard_reinit.cpp]: >/0 [](start = 54, length = 2)

Just search "/0"?

File: syncd/syncd_hard_reinit.cpp [hunk 2]
```diff
@@ -300,6 +301,82 @@ bool shouldSkipCreateion(
...
+    }
+    else
+    {
+        // default trap group
+        return;
+    }
+
+    sai_object_type_t objectType = SAI_OBJECT_TYPE_TRAP_GROUP;
+
+    SWSS_LOG_INFO("creating trap group and setting attributes 1 by 1 as workaroung");
```
Comment (lguohan) [inline syncd/syncd_hard_reinit.cpp]: >workaroung [](start = 72, length = 10)

workaround

File: syncd/syncd_hard_reinit.cpp [hunk 3]
```diff
@@ -717,15 +817,24 @@ sai_unicast_route_entry_t getRouteEntryFromString(const std::string &strRouteEnt
...
         const std::string &strRouteEntry = kv.first;
         const std::string &asicKey = kv.second;
 
+        bool isDefault = strRouteEntry.find("/0") != std::string::npos;
+
+        if (defaultOnly ^ isDefault)
+        {
+            continue;
+        }
+
```
Comment (lguohan) [inline syncd/syncd_hard_reinit.cpp]: now we need to enumerate all routes in the table twice? but it looks like you have benchmarked this function, so how long does it take for this function when defaultOnly is true?

File: syncd/syncd_reinit.cpp [hunk 4]
```diff
@@ -748,6 +748,182 @@ void helperCheckPriorityGroupsIds()
...
+                else
+                {
+                    SWSS_LOG_ERROR(
+                            "unexpected object type %s on scheduler group child list 0x%llx",
+                            sai_serialize_object_type(objectType).c_str(),
+                            schedGroupId);
+
+                    exit_and_notify(EXIT_FAILURE);
+                }
+            }
```
Comment (lguohan) [inline syncd/syncd_reinit.cpp]: this seems unnecessary, the childlist scheduler groups should be include in the get schedulergroup list

Comment (kcudnik) [inline syncd/syncd_reinit.cpp]: removed

**Pull request merged**